### DISCUSSION TITLE
Include for included relationship

### DISCUSSION THREAD

**Discussion by rehatkathuria** [prisma/prisma#21170] [Q&A] [answered]
### Question

Is it possible to fetch a deeply nested relationship? Say I have the following: 

```js
const foo = await prisma.foo.findMany({
  where: { user_id: user.id },
  orderBy: { created_at: "asc" },
  select: { user: true },
})
```

Is it possible for me to specify an `include` for a relationship that's on the `user` above?

**Comment by rehatkathuria** [comment]
I currently have the following and am wondering if it's possible to do with a single query:

```js
const foo = await prisma.foo.findMany({
		where: { user_id: user.id },
		orderBy: { created_at: "asc" },
		select: { user: true },
	})

	const users = interests.map((iterating) => {
		return iterating.user.id
	})

	const populated = await prisma.user.findMany({
		where: { id: { in: users } },
		include: { profile_assets: true },
	})
```

**Comment by ludralph** [comment, marked as answer]
Hi @rehatkathuria 👋 

 It is possible to fetch deeply nested relationships using Prisma. You can use the `include` option to include related records in the query response. You can also nest include options to include relations of relations. for example:
```js
const foo = await prisma.foo.findMany({
  where: { user_id: user.id },
  orderBy: { created_at: "asc" },
  include: { 
    user: {
      include: { 
        relatedField: true 
      }
    }
  },
})
```
For the second question, can you please share your schema meanwhile I think you can do something like this
```js
const foo = await prisma.foo.findMany({
  where: { user_id: user.id },
  orderBy: { created_at: "asc" },
  include: { 
    user: {
      include: { profile_assets: true },
    }
  },
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by rehatkathuria** [reply]
That works brilliantly, thank you @ludralph!
